### ISSUE TITLE
Java14 Full Records Support Check Validation: ModifierOrderCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8509]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_modifier.html#ModifierOrder

From check documentation:
_Checks that the order of modifiers conforms to the suggestions in the Java Language specification, § 8.1.1, 8.3.1, 8.4.3 and 9.4. The correct order is:_

- public
- protected
- private
- abstract
- default
- static
- final
- transient
- volatile
- synchronized
- native
- strictfp

```

➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="ModifierOrder"/>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
class TestClass {
    /** Illegal order of modifiers for variables */
    static private boolean sModifierOrderVar = false;

    /**
     * Illegal order of modifiers for methods. Make sure that the
     * first and last modifier from the JLS sequence is used.
     */
    strictfp private void doStuff() // violation
    {
    }

    /** Single annotation without other modifiers */
    @MyAnnotation2 void someMethod() // violation
    {
    }

    /** Illegal order of annotation - must come first */
    private @MyAnnotation2 void someMethod2() // violation
    {
    }

    /** Annotation in middle of other modifiers otherwise in correct order */
    private @MyAnnotation2 strictfp void someMethod3() // violation
    {
    }

}

record MyRecord1(boolean t, boolean f) {
    /** Illegal order of modifiers for variables */
    static private boolean sModifierOrderVar = false; // violation

    /**
     * Illegal order of modifiers for methods. Make sure that the
     * first and last modifier from the JLS sequence is used.
     */
    strictfp private void doStuff() // violation
    {
    }

    /** Single annotation without other modifiers */
    @MyAnnotation2 void someMethod()
    {
    }

    /** Illegal order of annotation - must come first */
    private @MyAnnotation2 void someMethod2() // violation
    {
    }

    /** Annotation in middle of other modifiers otherwise in correct order */
    private @MyAnnotation2 strictfp void someMethod3() // violation
    {
    }

}

@interface MyAnnotation2 {
}



➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:3:12: 'private' modifier out of order with the JLS suggestions. [ModifierOrder]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:9:14: 'private' modifier out of order with the JLS suggestions. [ModifierOrder]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:19:13: '@MyAnnotation2' annotation modifier does not precede non-annotation modifiers. [ModifierOrder]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:24:13: '@MyAnnotation2' annotation modifier does not precede non-annotation modifiers. [ModifierOrder]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:32:12: 'private' modifier out of order with the JLS suggestions. [ModifierOrder]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:38:14: 'private' modifier out of order with the JLS suggestions. [ModifierOrder]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:48:13: '@MyAnnotation2' annotation modifier does not precede non-annotation modifiers. [ModifierOrder]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:53:13: '@MyAnnotation2' annotation modifier does not precede non-annotation modifiers. [ModifierOrder]
Audit done.
Checkstyle ends with 8 errors.

```

This check works as intended within the body of a record definition, so I think we can close this one.

**Comment by pbludov** [comment]
I agree. This check handles `TokenTypes.MODIFIERS`, so this is already supported:
```java
class TestClass {
    static public strictfp record A() {}
    strictfp public static record B() {}
}
```
produces
```
Starting audit...
[ERROR] /home/pbludov/src/tmp/TestClass.java:2:12: 'public' modifier out of order with the JLS suggestions. [ModifierOrder]
[ERROR] /home/pbludov/src/tmp/TestClass.java:3:14: 'public' modifier out of order with the JLS suggestions. [ModifierOrder]
Audit done.
```
